addons/trenches/CfgVehicles.hpp

```diff
@@ -1,46 +1,50 @@
 class CBA_Extended_EventHandlers;
 
-#define ACE_TRENCHES_ACTIONS class ACE_Actions { \
-        class ACE_MainActions { \
-            displayName = ECSTRING(interaction,MainAction); \
-            selection = ""; \
-            distance = 3; \
-            condition = QUOTE(true); \
-            class ACE_ContinueDiggingTrench { \
-                displayName = CSTRING(ContinueDiggingTrench); \
-                condition = QUOTE([ARR_2(_target,_player)] call FUNC(canContinueDiggingTrench)); \
-                statement = QUOTE([ARR_2(_target,_player)] call FUNC(continueDiggingTrench);); \
-            }; \
-            class ACE_RemoveTrench { \
-                displayName = CSTRING(RemoveEnvelope); \
-                condition = QUOTE([ARR_2(_target,_player)] call FUNC(canRemoveTrench)); \
-                statement = QUOTE([ARR_2(_target,_player)] call FUNC(removeTrench);); \
-            }; \
-        }; \
-    }
-
 class CfgVehicles {
     class Man;
     class CAManBase: Man {
         class ACE_SelfActions {
             class ACE_Equipment {
-                class GVAR(digEnvelopeSmall) {
-                    displayName = CSTRING(DigEnvelopeSmall);
-                    condition = QUOTE(_player call FUNC(canDigTrench));
-                    //wait a frame to handle "Do When releasing action menu key" option
-                    statement = QUOTE([ARR_2({_this call FUNC(placeTrench)},[ARR_2(_this select 0,'ACE_envelope_small')])] call CBA_fnc_execNextFrame);
-                    exceptions[] = {};
-                    showDisabled = 0;
-                    //icon = QPATHTOF(UI\icon_sandbag_ca.paa);
-                };
-                class GVAR(digEnvelopeBig) {
-                    displayName = CSTRING(DigEnvelopeBig);
-                    condition = QUOTE(_player call FUNC(canDigTrench));
-                    //wait a frame to handle "Do When releasing action menu key" option
-                    statement = QUOTE([ARR_2({_this call FUNC(placeTrench)},[ARR_2(_this select 0,'ACE_envelope_big')])] call CBA_fnc_execNextFrame);
-                    exceptions[] = {};
+                class ACE_Trenches {
+                    displayName = CSTRING(ActionCategory);
+                    condition = QUOTE(_player call FUNC(canDigTrench) && GVAR(allowDigging));
+                    statement = "";
                     showDisabled = 0;
-                    //icon = QPATHTOF(UI\icon_sandbag_ca.paa);
+                    priority = 3;
+                    // icon = "z\ace\addons\trenches\UI\w_entrchtool_ca.paa";
+                    exceptions[] = {"notOnMap", "isNotInside", "isNotSitting"};
+                    class GVAR(digEnvelopeShort) {
+                       displayName = CSTRING(DigEnvelopeShort);
+                       condition = QUOTE(GVAR(allowDigging) && ([ARR_2(_target,_player)] call FUNC(canContinueDiggingTrench)) && GVAR(allowShortEnvelope));
```

Comment:
You can use `_this call FUNC` instead of the ugly `ARR_n` macro. Target and player might be the other way around though, but they will be first 2 parameters.

I would also suggest using single quotes instead of `QUOTE` macro, because all those `ARR_n` are really unreadable.

Lazy evaluation here and in other places might be beneficial as well.